dojo_plugin/pages/course.py

```diff
@@ -109,9 +121,58 @@ def query(module_id):
 
     module_solves = {}
 
+
+    def get_meme_progress(dojo, user_id):
+            discord_user =  DiscordUsers.query.where(DiscordUsers.user_id == user_id).first()
+            if not discord_user:
+                return ""
+            course_start = dojo.start_date()
+            meme_weeks = discord_user.meme_dates(start=dojo.start_date(), end=dojo.start_date() + datetime.timedelta(weeks=21))
+            week_ranges = ' '.join([f"{w[0].month}/{w[0].day}-{w[1].month}/{w[1].day}" for w in meme_weeks])
+            return week_ranges
+
+    def get_thanks_progress(dojo, user_id):
+            discord_user =  DiscordUsers.query.where(DiscordUsers.user_id == user_id).first()
+            if not discord_user:
+                return 0
+            thanks_count = discord_user.thanks_count(start=dojo.start_date())
+
+            return thanks_count
+
     def result(user_id):
         assessment_grades = []
 
+        ec_limit = dojo.course.get("ec_limit") or 1.00
```

Comment:
Should the ec_limit changes be made to the course.yml's before this PR goes live? There might be a brief period where EC has a different cap than what's expected, some student might see it and then get confused.

Reply:
That `ec_limit` should only really be needed by fall cse365 currently.  All other courses have individual ec type limits that can be set with the `max_credit` property on the type.